lib/teiserver/protocols/spring/spring_out.ex

```diff
@@ -737,6 +737,34 @@ defmodule Teiserver.Protocols.SpringOut do
       end
     end)
 
+    # Send BATTLETEAMS batch message
+    if Config.get_site_config_cache("lobby.Broadcast Battle Teams Information") do
+      ttl_ms = 1_000
+      now = System.system_time(:millisecond)
+      teams_data =
+        case Teiserver.cache_get(:application_temp_cache, :battle_teams) do
+          {cached, ts} when now - ts < ttl_ms ->
+            cached
+          _ ->
+            data =
+              Lobby.list_lobby_ids()
+              |> Task.async_stream(
```

Comment:
oh, didn't know about `async_stream`. In this case, I think I prefer `await_many`, since you're going to send messages to different genserver anyway, the scheduler will take care of the rest, and you don't have to come up with some arbitrary `max_concurrency` number.